akka-http-core/src/main/scala/akka/http/impl/settings/ClientConnectionSettingsImpl.scala

```diff
@@ -24,7 +24,8 @@ private[akka] final case class ClientConnectionSettingsImpl(
   requestHeaderSizeHint: Int,
   websocketRandomFactory: () ⇒ Random,
   socketOptions: immutable.Seq[SocketOption],
-  parserSettings: ParserSettings)
+  parserSettings: ParserSettings,
+  redirectSettings: akka.http.scaladsl.settings.ClientAutoRedirectSettings)
```

Comment:
> I mean, adding a method shouldn't impact the binary compatibility, should it?

It depends. Imagine you've added an abstract method (like we do here), to a class which users are meant to extend. Suddenly you broke their code when they try to upgrade.

In this case the classes are lined up such that we can indeed silence this error – as only we implement this interface.